lib/reline/key_stroke.rb

```diff
@@ -99,6 +108,37 @@ def expand(input)
 
   private
 
+  # returns match status of CSI/SS3 sequence and matched length
+  def match_unknown_escape_sequence(input)
+    idx = 0
+    esc = 27 # "\e".ord
+    return [:unmatched, nil] unless input[idx] == esc
+    idx += 1
+    idx += 1 if input[idx] == esc
+
+    case input[idx]
+    when nil
+      return [:matching, nil]
+    when 91 # == '['.ord
+      # CSI sequence
+      idx += 1
+      idx += 1 while idx < input.size && (0x30..0x3f).cover?(input[idx])
+      idx += 1 while idx < input.size && (0x20..0x2f).cover?(input[idx])
+      input[idx] ? [:matched, idx + 1] : [:matching, nil]
```

Comment:
`0x30..0x3f` and `0x20..0x2f` are from here https://en.wikipedia.org/wiki/ANSI_escape_code#Terminal_input_sequences

Here are the behavior observed in my macOS environment.
readline and phython(3.10.9) only supports CSI sequence that match these regexp patterns `\e\[\d?.` `\e\[1;[35].` `\e\[3;5.` `\e\[20.`
node supports `\e\e?\[\d{0,2}(;\d?)?.`
sh, bash, zsh, python(2.7) and php(8.2.0) supports `\e\[3?.`
emacs seems to supports `\e\e?\[(\d(;\d)?)?.`
Only node and emacs supports F5(`"\e[15~"`) and option+up(`"\e\e[A"`)

Reply:
Just a quick note: can we store them in constants for documentation purpose + so we don't need to initialise new range objects on every call?

Reply:
> (Range object with number literal is optimized. 10.times.map{(1..3).__id__}.uniq.size #=> 1)

That's my TIL for the day. Thanks for sharing!